packages/form-js-viewer/src/core/FieldFactory.js

```diff
@@ -60,14 +60,13 @@ export class FieldFactory {
       throw new Error(`binding path '${[...parentPath, ...path.split('.')].join('.')}' is already claimed`);
     }
 
-    // currently using a hard coded solution while we evaluate refactoring
-    // cf. https://github.com/bpmn-io/form-js/issues/1291
-    const shouldApplyLabel = applyDefaults && !(config.type === 'datetime') && config.label;
-
-    const field = config.create({
-      ...(shouldApplyLabel ? { label: config.label } : {}),
-      ...attrs,
-    });
+    const field = config.create(
+      {
+        ...(config.label ? { label: config.label } : {}),
```

Comment:
applyDefaults removed here as some long time ago, we made the change to serialize empty labels to '' instead of undefined (so that they would persist across loads). Hence this wasn't doing anything to properly formatted schemas.